shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -436,8 +437,20 @@ - (nullable id)accessibilityElementAtIndex:(NSInteger)index {
     return _semanticsObject;
   }
   SemanticsObject* child = [_semanticsObject children][index - 1];
-  if ([child hasChildren])
-    return [child accessibilityContainer];
+    
+    // This 'if' block handles adding accessibility support for the embedded platform view.
+    // We first check if the child is a semantic node for a platform view.
+    // If so, we add the platform view as accessibilityElements of the child.
+    shell::FlutterPlatformViewsController *flutterPlatformViewsController = _bridge.get()->flutter_platform_views_controller();
+    if (child.node.platformViewId > -1 && flutterPlatformViewsController) {
+        NSObject<FlutterPlatformView> *platformViewProtocolObject = flutterPlatformViewsController->GetPlatformViewByID(child.node.platformViewId);
+        UIView *platformView = [platformViewProtocolObject view];
+        child.accessibilityElements = @[platformView];
```

Comment:
What if child is not an accessibility container (i.e. isAccessibilityElement returns false)? Does this still work?

Reply:
I assume child with platformViewId set to not -1 is always an accessibility container?

Reply:
It currently isn't though?

https://github.com/flutter/engine/blob/906d684a7765cfe37b0edcdd167d3e319c7a22d4/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L160-L172

Reply:
added ```  // If the node is the placeholder for a platform view. We know it should be a container.
  if ([self node].platformViewId > -1) {
    return false;
  }``` to enforce the container to not be accessibility element if the platformViewId is there